### DISCUSSION TITLE
Suggestions for broader adoption

### DISCUSSION THREAD

**Discussion by Sytten** [TriliumNext/Trilium#6336] [Ideas] [answered]
I think you have an awesome software and I really love it. I have a couple of suggestions to allow it to be used by more people/less tech people. Some of those people in the community could help build):
- Information website: can be static with links to the latest binaries, but at least it gives some internet presence and better SEO
- Sponsor link in github or another way to make donations
- Signed builds on OSX: it does cost money though, but I am sure the community can find 110USD per year for that. I think it is necessary to publish on brew?
- Installers: (speaking for osx, I didn't check other OS) we usually expect a DMG. Since this is usually an annoying job, it could be automated with github actions or similar.
- (long term) Provide hosted servers: I am not sure if you want to keep it as a hobby, but there is a market for that. I myself looked at notion and decided against it because it doesn't have offline mode and it's not open source, but I would pay for an hosted solution (with client-side encryption if possible).

Anyway those are just suggestions, what is already there is really amazing! I plan to write a blog once I get more familiar with the software and I think other people should too.

**Comment by zadam** [comment]
Hello,

all your items are valid concerns, however they also all cost time:

* I recently tried to setup GitHub sponsor page but I got stuck on the tax details - I'm not very familiar with local tax laws
* Mac in general is for me difficult since I don't own one and Apple makes it legally very difficult to develop for Mac without buying a Mac. Mac builds have only [limited support](https://github.com/zadam/trilium/wiki/FAQ#mac-os-support).
* DMG - similar to above, last time I checked it's not possible to build DMG on non-Mac platform. Apart from simple zipped directory, we have a .deb for Debian based systems. Some other major formats / installers might be supported in the future.
* I don't have any commercial plans for the project. Apart from that I believe that being profitable in this space is not easy (Trilium is AGPL though so anybody can try it).

So in the end it comes down to the time which I'd rather spend on the core project itself even at the cost of smaller adoption. After all this is one of the major benefits of the hobby projects - you don't have to try to appeal to the widest possible user base.

But of course help with any of those items is welcome.

**Comment by Sytten** [comment]
- Never tried, but I trust you. Maybe a paypal link would be easier?
- I have a mac and I am happy to help better support the platform! I know some CI platforms allow mac builds (I think they have a cluster of mac mini under it to support that).
- Make sense! I think this could be hosted for not a lot of money on GCP Cloud Run. We basically would need to have a different storage backend and extract background jobs if any. Then anybody could have their instance and only pay for when they use it (probably less than 1$ per month for most people).

Make sense, I am already super happy with trilium and I plan to help once I get more comfortable with it.

**Comment by gramster** [comment]
Would it be fair to say that a lot of the Mac issues could be solved by having a different set of initial keyboard shortcut assignments? That seems like it could be a reasonably simple task. I saw you said you would prefer consistency across platforms but I don't think that is a realistic goal. You may instead be able to get similarity (e.g. JetBrains has different key maps for Mac/Linux/Windows but its not that hard to adapt between them).

**Comment by Sytten** [comment]
It certainly helps to have native shortcuts for productivity, some other suggestions are not related to that though.

**Comment by zadam** [comment]
I'm not necessarily against platform specific default shortcuts, but somebody would have to drive that effort.

**Comment by zadam** [comment]
I managed to figure out the tax thing and GitHub sponsor profile is now set up!

**Comment by leoossa** [comment]
Cool!
One more question - would sponsors have ie. more impact in terms of suggesting features or roadmap? Ie. I would like to have notion-like databases or at least something that could mimic this. Would sponsoring make it more possible to work on it?

**Comment by zadam** [comment]
Realistically no - I expect the individual donations to be mostly in the $1 - $5 range and see them just as a way to express support/appreciation with no strings attached.

**Comment by Myzel394** [comment]
> Information website: can be static with links to the latest binaries, but at least it gives some internet presence and better SEO

@zadam what do you think about this?

**Comment by Myzel394** [comment]
> Signed builds on OSX: it does cost money though, but I am sure the community can find 110USD per year for that. I think it is necessary to publish on brew?

I don't think that's possible. Most people don't sponsor anything at all, even though it would make an extreme difference if everyone just donated like 1$.

Node.js for windows for example is an application more well-known than Trilium Notes and even the authors of this tool don't get enough sponsors to pay for code signing (see example https://github.com/coreybutler/nvm-windows/releases)

**Comment by zadam** [comment]
@Myzel394 information website would be great, but for me it's just another distraction/time sink. It would be awesome, if somebody else would volunteer to create one/maintain it.

**Comment by Myzel394** [comment]
> information website would be great, but for me it's just another distraction/time sink. It would be awesome, if somebody else would volunteer to create one/maintain it.

I am willing to create an information website. I would probably create it with Nextjs and you could host it using Vercel. You basically just connect your GitHub account with it and boom - website is online (as long as you don't earn anything with it, the hosting is free, you probably need to check if donations are okay).

However, before I can start programming, we need to define what we would like to show on the page and we need to design it of course :D

So I would suggest creating a common landing page. So show what Trilium Notes is, what you can do with it and how you can use it.

**Comment by zadam** [comment]
> I would probably create it with Nextjs and you could host it using Vercel. 

Isn't this supposed to be a website - i.e. not a web app but a static web presentation?

**Comment by Myzel394** [comment]
You can also use Vercel to host static content. Vercel is the go-to platform if you're using Next.js (also for static content), but you can also use something like netlify for example for static content.

**Comment by zadam** [comment]
GitHub pages also allow you to host a static website. This might make it easy to manage it together with the repo ...

Why do you want to use Next.js for it? I mean in my mind this is a single HTML landing page with some nice CSS and few screenshots.

**Comment by sigaloid** [comment]
@Sytten: With regards to your fifth bullet ("Provide hosted servers"), I've created a relevant issue: #2496

**Comment by cwilliams5** [comment]
> GitHub pages also allow you to host a static website. This might make it easy to manage it together with the repo ...
> 
A webpage is a way I could give back to the project if there is still interest. If your interested zadam just let me know. I agree keep it github pages would be simple. Would us maintainers need some sort of access to commit to it? Any preference for a framework for development? 

Lastly github pages would support a custom domain. Looks like trilium.app or trilium.dev or trilium.info is still available, or maybe some other domain squatter would part with theirs for a low cost. I could snap one of those up if interested but I want zadam to be in control.

**Comment by sigaloid** [comment]
I've bought and set up https://trilium.cc and emailed zadam about it however I have not heard back 😔 

It's actually a shared note running on a Trilium instance!

**Comment by zadam** [comment]
@sigaloid yeah, sorry, I forgot to reply. But TBH I'm not sure what to do with that now.

**Comment by inson1** [comment]
@sigaloid Where is the repo for that?

**Comment by zadam** [comment]
Closing this one, if there are new ideas, post as a new discussion.

**Comment by eliandoran** [comment, marked as answer]
> Information website: can be static with links to the latest binaries, but at least it gives some internet presence and better SEO
Sponsor link in github or another way to make donations

Work in progress

> Signed builds on OSX: it does cost money though, but I am sure the community can find 110USD per year for that. I think it is necessary to publish on brew?

Done.

> Installers: (speaking for osx, I didn't check other OS) we usually expect a DMG. Since this is usually an annoying job, it could be automated with github actions or similar.

Done.

> (long term) Provide hosted servers: I am not sure if you want to keep it as a hobby, but there is a market for that. I myself looked at notion and decided against it because it doesn't have offline mode and it's not open source, but I would pay for an hosted solution (with client-side encryption if possible).

Done indirectly via PikaPods and trilium.cc.
